crates/sui-indexer-alt-graphql/src/api/types/transaction_kind/consensus_commit_prologue.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::Object;
+use fastcrypto::encoding::{Base58, Encoding};
+use sui_types::{
+    digests::{AdditionalConsensusStateDigest, ConsensusCommitDigest},
+    messages_consensus::{
+        ConsensusCommitPrologue as NativeConsensusCommitPrologueV1,
+        ConsensusCommitPrologueV2 as NativeConsensusCommitPrologueV2,
+        ConsensusCommitPrologueV3 as NativeConsensusCommitPrologueV3,
+        ConsensusCommitPrologueV4 as NativeConsensusCommitPrologueV4,
+    },
+};
+
+use crate::{
+    api::{
+        scalars::{date_time::DateTime, uint53::UInt53},
+        types::epoch::Epoch,
+    },
+    error::RpcError,
+    scope::Scope,
+};
+
+#[derive(Clone)]
+pub(crate) struct ConsensusCommitPrologueTransaction {
+    pub(crate) scope: Scope,
+    pub(crate) epoch: u64,
+    pub(crate) round: u64,
+    pub(crate) sub_dag_index: Option<u64>,
+    pub(crate) commit_timestamp_ms: u64,
+    pub(crate) consensus_commit_digest: Option<ConsensusCommitDigest>,
+    pub(crate) additional_state_digest: Option<AdditionalConsensusStateDigest>,
+}
+
+/// System transaction that runs at the beginning of a checkpoint, and is responsible for setting the current value of the clock, based on the timestamp from consensus.
+#[Object]
+impl ConsensusCommitPrologueTransaction {
+    /// Epoch of the commit prologue transaction.
+    async fn epoch(&self) -> Option<Epoch> {
+        Some(Epoch::with_id(self.scope.clone(), self.epoch))
+    }
+
+    /// Consensus round of the commit.
+    async fn round(&self) -> Option<UInt53> {
+        Some(self.round.into())
+    }
+
+    /// Unix timestamp from consensus.
+    async fn commit_timestamp(&self) -> Result<Option<DateTime>, RpcError> {
+        Ok(Some(DateTime::from_ms(self.commit_timestamp_ms as i64)?))
+    }
+
+    /// Digest of consensus output, encoded as a Base58 string (only available from V2+ of the transaction).
+    async fn consensus_commit_digest(&self) -> Option<String> {
+        self.consensus_commit_digest
+            .as_ref()
+            .map(|digest| Base58::encode(digest.inner()))
+    }
+
+    /// Sub-DAG index for consensus ordering (only available from V3+ of the transaction).
+    async fn sub_dag_index(&self) -> Option<UInt53> {
+        self.sub_dag_index.map(|idx| idx.into())
+    }
+
+    /// Additional state digest for enhanced security (only available from V4+ of the transaction).
+    async fn additional_state_digest(&self) -> Option<String> {
+        self.additional_state_digest
+            .as_ref()
+            .map(|digest| digest.to_string())
+    }
+}
```

Comment:
Let's align with the documentation from gRPC here, it seems more thorough:

https://github.com/MystenLabs/sui/blob/cd58464d654d43c23d3615d69826bf1a58ce61c3/crates/sui-rpc-api/proto/sui/rpc/v2beta/transaction.proto#L298-L338